jobs update stanza merging cleanup

langmartin opened the pull request:
Add a test that demonstrates that update stanza merging, and make AutoPromote match AutoRevert

- [ ] Changelog

notnoop commented on api/jobs.go:
```diff
@@ -513,10 +512,6 @@ func (u *UpdateStrategy) Canonicalize() {
 
-	if u.AutoRevert == nil {
-		u.AutoRevert = d.AutoRevert
-	}
```
Not sure I follow the reason for this removal.  I might be missing some context.  Also, if this is a bug, can we have a test case please?

langmartin replied:
there's a comment out of view at the top of the struct that explains the reason: with booleans in the default struct, a missing value is masked by a false, which seems like an intentional false when merging update stanzas together

notnoop replied:
I see, thanks.  It would be useful to point to the comment that in the PR description to aid the reviewer, who isn't as familiar with the code.  Also, I don't quite understand it: If user leaves field unset in config but the default value of the field is `true`, I would have expected `Canonicalize` to render the field as `true`?   Where else do apply default values?

Also, the test is failing.

langmartin commented:
I added another test and dialed back the scope on this, merging is working for AutoRevert already, and this is now just making AutoPromote work the same way. I've updated the description too.
